app/experimenter/experiments/forms.py

```diff
@@ -266,47 +175,135 @@ def clean_population_percent(self):
         return population_percent
 
 
+class ExperimentVariantAddonForm(NameSlugMixin, forms.ModelForm):
+
+    experiment = forms.ModelChoiceField(
+        queryset=Experiment.objects.all(), required=False
+    )
+    is_control = forms.BooleanField(required=False)
+    ratio = forms.IntegerField(
+        label="Branch Size",
+        initial="50",
+        help_text=Experiment.CONTROL_RATIO_HELP_TEXT,
+        widget=forms.TextInput(attrs={"class": "form-control"}),
+    )
+    name = forms.CharField(
+        label="Name",
+        help_text=Experiment.CONTROL_NAME_HELP_TEXT,
+        widget=forms.TextInput(attrs={"class": "form-control"}),
+    )
+    description = forms.CharField(
+        label="Description",
+        help_text=Experiment.CONTROL_DESCRIPTION_HELP_TEXT,
+        widget=forms.Textarea(attrs={"class": "form-control", "rows": 3}),
+    )
+    slug = forms.CharField(required=False)
+
+    class Meta:
+        model = ExperimentVariant
+        fields = [
+            "description",
+            "experiment",
+            "is_control",
+            "name",
+            "ratio",
+            "slug",
+        ]
+
+
+class ExperimentVariantPrefForm(ExperimentVariantAddonForm):
+
+    value = JSONField(
+        label="Pref Value",
+        help_text=Experiment.CONTROL_VALUE_HELP_TEXT,
+        widget=forms.TextInput(attrs={"class": "form-control"}),
+    )
+
+    class Meta:
+        model = ExperimentVariant
+        fields = [
+            "description",
+            "experiment",
+            "is_control",
+            "name",
+            "ratio",
+            "slug",
+            "value",
+        ]
+
+
+class ExperimentVariantsFormSet(BaseInlineFormSet):
+
+    def clean(self):
+        total_percentage = sum(
+            [
+                form.cleaned_data["ratio"]
+                for form in self.forms
+                if not form.cleaned_data["DELETE"]
+            ]
+        )
+
+        if total_percentage != 100:
+            for form in self.forms:
+                form._errors["ratio"] = [
+                    "The size of all branches must add up to 100"
+                ]
+
+        unique_slugs = set([form.cleaned_data["slug"] for form in self.forms])
```

Comment:
Does this also account for deleted forms?

Reply:
Lol we realized that at the same time and I updated it